docs/cli.md

```diff
@@ -12,33 +12,112 @@ To build the CLI, follow these steps:
 
 ## Usage
 
-#### Set Agent URL
-```shell
-export AGENT_GRPC_URL=<agent_host:agent_host>
+#### Generate Keys
+
+To generate a public & private key pair, run the following command:
+
+```bash
+./build/cocos-cli keys
+```
+
+This will generate a key pair of type rsa. Different key types can be generated using the `-k` flag. Currently supported types on cocos are rsa, ecdsa and ed25519.
+
+```bash
+./build/cocos-cli keys -k ecdsa
 ```
 
 #### Upload Algorithm
 
 To upload an algorithm, use the following command:
 
 ```bash
-./build/cocos-cli algo /path/to/algorithm
+export AGENT_GRPC_URL=<agent_host:agent_host>
+
+./build/cocos-cli algo /path/to/algorithm /path/to/private/key
 ```
 
+Currently, support is provided for two types of algorithms: executable binaries and python files. The above command expects an algorithm in the binary format which will be executed inside agent. For python files, the algo file is required, along with the requirements file and the python runtime. To run a python file, use the following command:
+
+```bash
+export AGENT_GRPC_URL=<agent_host:agent_host>
+
+./build/cocos-cli algo /path/to/algorithm /path/to/private/key --algorithm python --requirements /path/to/requirements.txt --python-runtime python
+```
+
+The agent grpc url is required for this operation, this will be available once the TEE has been provisioned and agent is running.
```

Comment:
move this and export to the beginning so you don't have to repeat

Reply:
I added it there since it is not required for all cli functions.

Reply:
just have at the beginning of the cli examples and write that note that it is only required for certain functions. or rather add a section for env variables used by cli instead